src/SnapKeyring.ts

```diff
@@ -397,68 +398,173 @@ export class SnapKeyring extends EventEmitter {
     chainId?: string;
   }): Promise<Json> {
     const { account, snapId } = this.#resolveAddress(address);
-    if (!account.methods.includes(method as EthMethod)) {
-      throw new Error(
-        `Method '${method}' not supported for account ${account.address}`,
-      );
-    }
+    this.#validateMethod(method as EthMethod, account);
```

Comment:
I would go for `hasMethod` instead, and maybe have the `account` as a first argument (more like if we would use `account` as an object/receiver for the method).

This is purely cosmetic, so you can disagree 🙈

Reply:
I like it 😄
Implemented in this commit [0522eb3](https://github.com/MetaMask/eth-snap-keyring/pull/232/commits/0522eb325e4abf8bfc279b1891737872fda20f3a)